src/typing/env_js.ml

```diff
@@ -75,33 +80,58 @@ let find_env ?(for_type=false) cx x reason =
   let rec loop = function
     | [] -> find_global cx x reason
     | block::blocks ->
-        (match SMap.get x !block with
+        let entries = block.entries in
+        (match SMap.get x !entries with
         | Some entry when (not entry.for_type || for_type) -> block
         | _ -> loop blocks)
   in
   loop (!env)
 
 let get_block x block =
-  SMap.find_unsafe x !block
+  let entries = block.entries in
+  SMap.find_unsafe x !entries
 
 let set_block x entry block =
-  block := !block |> SMap.add x entry
+  let entries = block.entries in
+  entries := !entries |> SMap.add x entry
 
 let exists_block x block =
-  SMap.get x !block
+  let entries = block.entries in
+  SMap.get x !entries
+
+let set_env x entry hoist =
```

Comment:
`set_in_env` and maybe use the `scope_kind` variant here (rather than a bool) so that it's more immediately clear what callsites are doing when they call this function (rather than passing `true` or `false` which could mean anything until you go and find and read the function definition).

i.e. `set_in_env "myVar" var_entry HoistedScope` or